### PR TITLE
Add a few missing methods/fields to AudioContext

### PR DESCRIPTION
**Pull request by ramnivas**
Add the close(), suspend(), and resume() methods as well
as the state field to the AudioContext.

### CODE REVIEW

File: src/main/scala/org/scalajs/dom/raw/Audio.scala [hunk 1]
```diff
@@ -47,6 +47,14 @@ class AudioContext extends EventTarget {
    */
   val sampleRate: Double = js.native
 
+  /** Returns the current state of the AudioContext. */
+  val state: String = js.native
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/raw/Audio.scala]: I guess this should be a `def`.

File: src/main/scala/org/scalajs/dom/raw/Audio.scala [hunk 2]
```diff
@@ -47,6 +47,14 @@ class AudioContext extends EventTarget {
...
+
+  /** Closes the audio context, releasing any system audio resources
+   * that it uses.
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/raw/Audio.scala]: missing space before `that`

**Comment by sjrd** [comment]
Could you give a link to the spec of these members?

**Comment by ramnivas** [comment]
@sjrd Link to the spec: https://developer.mozilla.org/en-US/docs/Web/API/AudioContext

File: src/main/scala/org/scalajs/dom/raw/Audio.scala [hunk 3]
```diff
@@ -47,6 +47,14 @@ class AudioContext extends EventTarget {
    */
   val sampleRate: Double = js.native
 
+  /** Returns the current state of the AudioContext. */
+  def state: String = js.native
+
+  /** Closes the audio context, releasing any system audio resources
+   *  that it uses.
+   */
+  def close(): Unit = js.native
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/raw/Audio.scala]: From the spec, this also returns a `js.Promise[Unit]`.

**Comment by sjrd** [comment]
That's all.

**Comment by mseddon** [comment]
Not around atm, but if it corresponds to [this](https://www.w3.org/TR/2015/WD-webaudio-20151208/) (or a previous version) then it should be good.  MDN can be a little out of date at times.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**